### PR TITLE
Fixed bug for full-model derivative approximation when desvars or responses have indices specified.

### PR DESCRIPTION
**Pull request by Kenneth-T-Moore**

### CODE REVIEW

File: openmdao/core/system.py [hunk 1]
```diff
@@ -191,10 +191,14 @@ class System(object):
         If True, this system approximated its Jacobian
     _owns_approx_jac_meta : dict
         Stores approximation metadata (e.g., step_size) from calls to approx_total_derivs
-    _owns_approx_wrt : set or None
-        Overrides aproximation inputs.
     _owns_approx_of : set or None
         Overrides aproximation outputs.
+    _owns_approx_of_idx : dict
```
Comment (JustinSGray) [inline openmdao/core/system.py]: these descriptions are a little vague. Is the override only happening in the context of FD? Could you make them a bit more clear?

Comment (Kenneth-T-Moore) [inline openmdao/core/system.py]: Fixed

File: openmdao/core/tests/test_approx_derivs.py [hunk 2]
```diff
@@ -618,6 +618,63 @@ def test_sellar(self, vec_class):
...
+                # Unknowns
+                self.add_output('y1', np.zeros([4]))
+
+            def compute(self, inputs, outputs):
+                """
+                Execution.
+                """
+                outputs['y1'] = self.JJ.dot(inputs['x1'])
+
+            def compute_partials(self, inputs, outputs, partials):
```
Comment (JustinSGray) [inline openmdao/core/tests/test_approx_derivs.py]: this needs to be updated to the latest api now, right?

File: openmdao/core/tests/test_check_derivs.py [hunk 3]
```diff
@@ -557,6 +557,98 @@ def test_cs(self):
...
+                # Unknowns
+                self.add_output('y1', np.zeros([4]))
+
+            def compute(self, inputs, outputs):
+                """
+                Execution.
+                """
+                outputs['y1'] = self.JJ.dot(inputs['x1'])
+
+            def compute_partials(self, inputs, outputs, partials):
```
Comment (JustinSGray) [inline openmdao/core/tests/test_check_derivs.py]: update to latest api

Comment (Kenneth-T-Moore) [inline openmdao/core/tests/test_check_derivs.py]: Done

**Pull request merged**